docs/core/compatibility/core-libraries/6.0/static-virtual-interface-methods.md

```diff
@@ -0,0 +1,41 @@
+---
+title: "Breaking change: Static virtual methods in interfaces"
+description: Learn about the .NET 6.0 breaking change where `static` interface methods can now be marked `virtual`.
+ms.date: 09/08/2021
+---
+# Static virtual methods in interfaces
+
+.NET 6 is previewing a new feature where `static` interface methods can be marked as `virtual`. This feature involves several changes to the ECMA spec to allow intermediate language (IL) metadata patterns that were previously considered illegal. For more information, see [dotnet/runtime#49558](https://github.com/dotnet/runtime/issues/49558).
+
+> [!NOTE]
+> The feature is in preview, and it's possible that additional changes or support may be introduced between .NET 6 and a future version of .NET when the feature stabilizes.
+
+## Old behavior
+
+Static interface methods could not be marked as `virtual` and would be considered illegal IL.
```

Comment:
This is my mistake in the original issue, but this should be `abstract` based on what the language supports in the .NET 6 preview.
```suggestion
Static interface methods could not be marked as `abstract` and would be considered illegal IL.
```

Reply:
Notably, C# requires `abstract` but the IL metadata considers this to be `virtual` with no provided implementation. The concepts are the same, but differ based on whether you're considering this from the language or metadata perspective.

Reply:
> Notably, C# requires `abstract` but the IL metadata considers this to be `virtual` with no provided implementation. The concepts are the same, but differ based on whether you're considering this from the language or metadata perspective.

Are we confusing what is "legal" (i.e. what can be handled by ILAsm) with what is "legal" in metadata (i.e. in tables)?

Reply:
We're only considering what the JIT considers to be valid IL here.

Prior to .NET 6, the runtime would fail to load a type where the metadata marked a `static` method as `virtual`
In .NET 6, the runtime allows this in interfaces provided there is no implementation
In .NET vFuture, the runtime will likely allow an implementation to also be provided (to tie into DIMs)